plugins/modules/cloud/misc/terraform.py

```diff
@@ -368,7 +368,7 @@ def main():
     if needs_application and not module.check_mode and not state == 'planned':
         rc, out, err = module.run_command(command, cwd=project_path)
         # checks out to decide if changes were made during execution
-        if '0 added, 0 changed' not in out and not state == "absent" or '0 destroyed' not in out:
+        if ' 0 added, 0 changed' not in out and not state == "absent" or ' 0 destroyed' not in out:
```

Comment:
```suggestion
        if ' 0 added, 0 changed, 0 destroyed' not in out and not state == "absent" or ' 0 destroyed' not in out:
```

Reply:
There are cases where `terraform apply` will solely destroy resources.

Reply:
I agree there are times when `terraform apply` will only destroy resources, however the `or ' 0 destroyed' not in out:` part of the test already covers this scenario as it is executed regardless of state being `present` or `absent`. 

I have tested this to confirm that the `terraform` module registers a change when solely destroying resources via a `terraform apply`.

```
TASK [Debug terraform results] *********************************************************************************************************************************************************************************
ok: [localhost] => {
    "tf_outputs": {
        "changed": true,
        "command": "/usr/local/bin/terraform apply -no-color -input=false -auto-approve=true -lock=true /tmp/tmpzp4aidb9.tfplan",
        "failed": false,
        "outputs": {
            "mgmt_address": {
                "sensitive": false,
                "type": "string",
                "value": "RE.DA.CT.ED"
            }
        },
        "state": "present",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "google_compute_instance.bastion_instances[\"europe-west2-a\"]: Destroying... [id=projects/redacted-redacted-redacted/zones/europe-west2-a/instances/bastion-just-jackass]\ngoogle_compute_instance.bastion_instances[\"europe-west2-a\"]: Still destroying... [id=projects/redacted-redacted-redacted/zones/e...west2-a/instances/bastion-just-jackass, 10s elapsed]\ngoogle_compute_instance.bastion_instances[\"europe-west2-a\"]: Still destroying... [id=projects/redacted-redacted-redacted/zones/e...west2-a/instances/bastion-just-jackass, 20s elapsed]\ngoogle_compute_instance.bastion_instances[\"europe-west2-a\"]: Destruction complete after 22s\n\nApply complete! Resources: 0 added, 0 changed, 1 destroyed.\n\nOutputs:\n\nmgmt_address = RE.DA.CT.ED\n",
        "stdout_lines": [
            "google_compute_instance.bastion_instances[\"europe-west2-a\"]: Destroying... [id=projects/redacted-redacted-redacted/zones/europe-west2-a/instances/bastion-just-jackass]",
            "google_compute_instance.bastion_instances[\"europe-west2-a\"]: Still destroying... [id=projects/redacted-redacted-redacted/zones/e...west2-a/instances/bastion-just-jackass, 10s elapsed]",
            "google_compute_instance.bastion_instances[\"europe-west2-a\"]: Still destroying... [id=projects/redacted-redacted-redacted/zones/e...west2-a/instances/bastion-just-jackass, 20s elapsed]",
            "google_compute_instance.bastion_instances[\"europe-west2-a\"]: Destruction complete after 22s",
            "",
            "Apply complete! Resources: 0 added, 0 changed, 1 destroyed.",
            "",
            "Outputs:",
            "",
            "mgmt_address = RE.DA.CT.ED"
        ],
        "workspace": "default"
    }
}
```